### PR TITLE
Fix CI and Build Badges

### PR DESCRIPTION
**Pull request by cdr-chakotay**
- [x] Fix Gradle Issues
- [x] Update Gradle
- [x] Fix Build scripts
- [x] Remove deprecated commands
- [x] Higher the API Levels in order to be part of the Google Specification
- [x] Remove Travis CI 
- [x] Replace With GH actions
- [x] Add release workflow

### CODE REVIEW

**Comment by cdr-chakotay** [comment]
Please don't merge yet, its still WIP

File: build.gradle [hunk 1]
```diff
@@ -2,25 +2,27 @@
...
+}
+
 allprojects {
     repositories {
-        jcenter()
         maven { url "https://jitpack.io" }
         maven {
             url 'https://maven.google.com'
         }
+        google()
```
Comment (Acconut) [inline build.gradle:23]: I think `google()` does the same as the `maven { url 'https://maven.google.com' }` above. Do we need both?

File: examples/build.gradle [hunk 2]
```diff
@@ -1,18 +1,19 @@
-apply plugin: 'com.android.application'
+apply plugin: 'com.android.library'
```
Comment (Acconut) [inline examples/build.gradle]: This change looks suspicious. The examples directory contains a standalone application, so I would not expect the plugin type to be changed to library.

File: examples/build.gradle [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
 
-android {
-    compileSdkVersion 24
-    buildToolsVersion '26.0.2'
+repositories {
+    mavenCentral()
+    jcenter()
```
Comment (Acconut) [inline examples/build.gradle:5]: Do we still need jcenter here? Since you also removed it in the other Gradle files.

File: examples/build.gradle [hunk 4]
```diff
@@ -23,9 +24,10 @@ android {
...
-    compile fileTree(dir: 'libs', include: ['*.jar'])
-    compile 'com.android.support:appcompat-v7:24.2.1'
-    compile 'com.android.support.constraint:constraint-layout:1.0.2'
-    compile 'com.transloadit.android.sdk:transloadit-android:0.0.2'
-    testCompile 'junit:junit:4.12'
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation 'androidx.appcompat:appcompat:1.0.0'
+    implementation 'com.android.support.constraint:constraint-layout:2.0.4'
+    implementation 'com.transloadit.sdk:transloadit:0.1.6'
+    implementation project(path: ':transloadit-android')
```
Comment (Acconut) [inline examples/build.gradle]:
```
implementation 'com.transloadit.sdk:transloadit:0.1.6'
implementation project(path: ':transloadit-android')
```

IIUC, this includes the Android SDK locally and from the repository. Is this duplication wanted?

File: transloadit-android/build.gradle [hunk 5]
```diff
@@ -1,23 +1,29 @@
...
-apply plugin: 'com.github.dcendents.android-maven'
+apply plugin: 'maven-publish'
 
 group 'com.transloadit.android.sdk'
 version = '0.0.4'
 description = "An Android Integration of the Transloadit's(https://transloadit.com) file uploading and encoding service."
 
+repositories {
+    mavenCentral()
+    jcenter()
```
Comment (Acconut) [inline transloadit-android/build.gradle:9]: Do we still need jcenter here?

**Review by Acconut — COMMENTED** [review]
Nice work so far! Left already a few comments

File: examples/build.gradle (hunk 3 above)
Comment (cdr-chakotay) [reply examples/build.gradle:5]: Yes, because tus android is still on Jcenter

File: build.gradle (hunk 1 above)
Comment (cdr-chakotay) [reply build.gradle:23]: I'll try out thank you :)

File: examples/build.gradle (hunk 2 above)
Comment (cdr-chakotay) [reply examples/build.gradle]: Ye,  you are right. That one might slipped me

File: examples/build.gradle (hunk 4 above)
Comment (cdr-chakotay) [reply examples/build.gradle]: Kind of, as it is not the same repo. But I got rid of it now :)

File: transloadit-android/build.gradle (hunk 5 above)
Comment (cdr-chakotay) [reply transloadit-android/build.gradle:9]: Yes, because of the tus Android client:

**Comment by cdr-chakotay** [comment]
@Acconut I would need some help with two things:

- [ ] We need a Slack Webhook secret / token

- [ ] Can you please remove Travis CI - App from Settings → Integrations ? I do not have enough rights to do that.

- [x] We would also need the Sonatype credentials

**Comment by cdr-chakotay** [comment]
The badge on this branch's README is the one for the website @kvz
[https://github.com/transloadit/android-sdk/tree/Fix_CI](https://github.com/transloadit/android-sdk/tree/Fix_CI)

or
[CI](https://github.com/transloadit/android-sdk/actions/workflows/CI.yml)

```
[![CI](https://github.com/transloadit/android-sdk/actions/workflows/CI.yml/badge.svg?branch=master)](https://github.com/transloadit/android-sdk/actions/workflows/CI.yml)
```

**Comment by Acconut** [comment]
> @Acconut I would need some help with two things:
> *  We need a Slack Webhook secret / token
> * Can you please remove Travis CI - App from Settings → Integrations ? I do not have enough rights to do that.

I also don' t have the access to these settings on GitHub and Slack. @kvz can you please setup those or give us access to do it ourselves?

**Comment by kvz** [comment]
Seems you are a Slack workspace admin Marius, and repo admin both. Does not suffice? If not please give me exact instructions via signal

**Comment by cdr-chakotay** [comment]
I'd be ready for a dry run regarding releases (it will only send it to Sonatype at this point and not release it automatically).
I would activate full automation, when we see it is working fine in production.

@Acconut Ready to merge if you are :)

**Review by Acconut — APPROVED** [review]

**Comment by Acconut** [comment]
> We need a Slack Webhook secret / token

I will get one. Until that we can already merge this, I think :)

> Can you please remove Travis CI - App from Settings → Integrations ? I do not have enough rights to do that.

I don´t think this is actually necessary. Removing the `.travis.yml` file is enough to disable Travis CI (I am not even sure if Travis is actually running our jobs anymore). I can´t remove Travis CI from the integrations list because it is installed on the organization level (i.e. github.com/transloadit) and for that I don´t have the rights. But we should be fine anyways.

> Ready to merge if you are :)

Great, let's try it!

**Pull request merged**